pytests/CMakeLists.txt

```diff
@@ -14,7 +14,7 @@ configure_file(
 )
 
 add_custom_target(check
-    COMMAND pytest
+    COMMAND pytest -x
```

Comment:
Let's stop after first failure, saves a lot of time while dev testing

Reply:
This is indeed very handy during development, but we would like to see all failures of a current commit in CI logs. Can you add add another target, `check_all` that doesn't pass this option and update the CI scripts to use that target instead?

Or better, can you work your way such that this flag isn't passed in CI based on some docker ENV? Let's not merge this as is.

Reply:
Never mind, I'll remove it. I thought what's the point of continuing when we hit an error, we need to fix all failures anyway. Like compilation errors, fix one by one.